packages/compass-sidebar/src/components/multiple-connections/sidebar.tsx

```diff
@@ -486,33 +182,57 @@ export function MultipleConnectionSidebar({
           activeWorkspace={activeWorkspace}
           filterRegex={activeConnectionsFilterRegex}
           onFilterChange={onActiveConnectionFilterChange}
-          onConnect={onConnect}
-          onNewConnection={onNewConnectionOpen}
-          onEditConnection={onEditConnection}
-          onRemoveConnection={onRemoveConnection}
-          onDuplicateConnection={onDuplicateConnection}
+          onConnect={(connectionInfo) => {
+            void connect(connectionInfo);
+          }}
+          onNewConnection={createNewConnection}
+          onEditConnection={(connectionInfo) => {
+            editConnection(connectionInfo.id);
+          }}
+          onRemoveConnection={(connectionInfo) => {
+            void removeConnection(connectionInfo.id);
+          }}
+          onDuplicateConnection={(connectionInfo) => {
+            void duplicateConnection(connectionInfo.id);
+          }}
           onCopyConnectionString={onCopyConnectionString}
-          onToggleFavoriteConnectionInfo={onToggleFavoriteConnectionInfo}
+          onToggleFavoriteConnectionInfo={(connectionInfo) => {
+            void toggleConnectionFavoritedStatus(connectionInfo.id);
+          }}
           onOpenConnectionInfo={onOpenConnectionInfo}
-          onDisconnect={onDisconnect}
+          onDisconnect={(id) => {
+            void disconnect(id);
+          }}
           onOpenCsfleModal={onOpenCsfleModal}
-          onOpenNonGenuineMongoDBModal={onOpenNonGenuineMongoDBModal}
-        />
-        <ConnectionFormModal
-          isOpen={isConnectionFormOpen}
-          setOpen={onNewConnectionToggle}
-          onCancel={onNewConnectionClose}
-          onSaveAndConnectClicked={onSaveAndConnectClicked}
-          key={activeConnectionId}
-          onSaveClicked={onSaveConnectionClicked}
-          initialConnectionInfo={activeConnectionInfo}
-          connectionErrorMessage={connectionErrorMessage}
-          preferences={formPreferences}
-        />
-        <NonGenuineWarningModal
-          isVisible={genuineMongoDBModalVisible}
-          toggleIsVisible={setGenuineMongoDBModalVisible}
+          onOpenNonGenuineMongoDBModal={(connectionId: string) => {
+            showNonGenuineMongoDBWarningModal(connectionId);
+          }}
         />
+        {editingConnectionInfo && (
+          <ConnectionFormModal
```

Comment:
Technically could've moved the form to compass-connections too already, but all the tests are also here and I'm trying to limit this already massive PR 😅